pkg/kube/util.go

```diff
@@ -83,10 +99,16 @@ func CreateClientset(kubeconfig, context string, fns ...func(*rest.Config)) (*ku
 }
 
 // DefaultRestConfig returns the rest.Config for the given kube config file and context.
-func DefaultRestConfig(kubeconfig, configContext string, fns ...func(*rest.Config)) (*rest.Config, error) {
-	config, err := BuildClientConfig(kubeconfig, configContext)
-	if err != nil {
-		return nil, err
+func DefaultRestConfig(ns string, kubeconfig, configContext string, fns ...func(*rest.Config)) (*rest.Config, error) {
+	remoteConfig, err := configFromRemoteKubeConfig(ns)
+	var config *rest.Config
+	if err == nil && remoteConfig != nil {
```

Comment:
This is going to be impossible to debug if there are unexpected errors, as the error is swallowed. And we cannot just log all of the errors because in some (actually, almost all) cases it is expected to fail. This goes in hand with my previous comment that we should not be mixing multicluster here.